frontend/src/app/sessions/user-sessions-wrapper/active-sessions/file-browser-dialog/file-browser-dialog.component.html

```diff
@@ -3,117 +3,124 @@
-    </div>
-  </div>
+    @if (session.download_in_progress) {
+      <div class="collab-card mb-2">
+        <p>Your download is being prepared…</p>
+        <mat-progress-bar class="mt-2" mode="indeterminate"></mat-progress-bar>
+      </div>
+    }
 
-  <mat-checkbox
-    *ngIf="!loadingFiles"
-    [formControl]="showHiddenFiles"
-    (change)="loadFiles()"
-    id="hidden-files"
-    >Show hidden files</mat-checkbox
-  >
+    <mat-checkbox
+      class="ml-1"
+      [formControl]="showHiddenFiles"
+      (change)="loadFiles()"
+      >Show hidden files</mat-checkbox
+    >
+    <hr />
+  }
 
   <mat-tree
     [dataSource]="dataSource"
-    [treeControl]="treeControl"
-    class="file-tree"
+    [childrenAccessor]="childrenAccessor"
+    class="max-h-[50vh] overflow-y-auto py-2"
   >
     <mat-tree-node
       *matTreeNodeDef="let node"
-      matTreeNodeToggle
       matTreeNodePadding
+      class="!h-8 !min-h-8"
     >
-      <mat-icon class="file-icon"> insert_drive_file </mat-icon>
+      <mat-icon class="mx-3 text-amber-500">insert_drive_file</mat-icon>
 
-      <p style="margin: 0px" [ngClass]="node.isNew ? 'new-file' : ''">
+      <p [ngClass]="node.isNew ? 'text-green-700' : ''">
         {{ node.name }}
       </p>
-      <div *ngIf="node.isNew">
+      @if (node.isNew) {
         <button mat-icon-button (click)="removeFile(node.path, node.name)">
           <mat-icon>cancel</mat-icon>
         </button>
-      </div>
+      }
     </mat-tree-node>
 
-    <mat-nested-tree-node *matTreeNodeDef="let node; when: hasChild">
-      <div class="mat-tree-node">
-        <div>
-          <button
-            mat-icon-button
-            matTreeNodeToggle
-            [attr.aria-label]="'Expand ' + node.name"
-          >
-            <mat-icon class="dir-icon"> folder </mat-icon>
-          </button>
-          {{ node.name }}
-        </div>
-        <button
-          mat-icon-button
-          color="primary"
-          (click)="download(node.path)"
-          [disabled]="session.download_in_progress"
-        >
-          <mat-icon>download_file</mat-icon>
-        </button>
-        <div>
-          <input
-            hidden
-            type="file"
-            #fileInput
-            (change)="addFiles(fileInput.files, node.path, node)"
-            multiple
-          />
-
-          <button mat-icon-button color="primary" (click)="fileInput.click()">
-            <mat-icon>upload_file</mat-icon>
-          </button>
-        </div>
+    <mat-tree-node
+      *matTreeNodeDef="let node; when: hasChild"
+      matTreeNodePadding
+      [isExpanded]="node.isExpanded"
+      [isExpandable]="true"
+      class="!h-8 !min-h-8"
+    >
+      <button
+        mat-icon-button
+        (click)="node.isExpanded = !node.isExpanded"
+        [attr.aria-label]="'Expand ' + node.name"
+        data-testid="folder-button"
+      >
+        <mat-icon class="text-blue-900">folder</mat-icon>
```

Comment:
I think making this change into `folder_open` when the node is opened could look nice

Reply:
Makes sense and was easy to implement using the new `node.isExpanded` attribute 👍